### PR TITLE
Qt6

### PR DESCRIPTION
**Pull request by tstenner**
~The default is still Qt5, unless the target is Android.~
Qt6 is required for Android and deploying with Qt6 needs the most recent commit of liblsl.
When Qt6 isn't found, everything still builds against Qt5, so there's not much changed unless you're actively porting to Qt6 or trying to target Android.

As long as Ubuntu 18.04 is still around, we'll need to support building against Qt5, but packaging apps for Android has gotten a lot better with Qt6.

Ubuntu 16.04 will be reaching its EOL next month, so in the interest of not having lots of `#ifdef`s for long deprecated features removed in Qt6 the PR also drops CI builds for Ubuntu 16.04.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,22 +1,19 @@
...
-	HINTS ${LSL_INSTALL_ROOT}
-	"${CMAKE_CURRENT_LIST_DIR}/../../LSL/liblsl/build/"
-	"${CMAKE_CURRENT_LIST_DIR}/../../LSL/liblsl/build/install"
-	"${CMAKE_CURRENT_LIST_DIR}/../../LSL/liblsl/out/build/x64-Release"
-	"${CMAKE_CURRENT_LIST_DIR}/../../LSL/liblsl/out/install/x64-Release"
-	PATH_SUFFIXES share/LSL
-)
+set(QT6_FIRST OFF)
+if(ANDROID)
+	set(QT5_FIRST ON)
```
Comment (cboulay) [inline CMakeLists.txt:11]: Doesn't Android want QT6_FIRST?

Comment (tstenner) [reply CMakeLists.txt]: Yes, I had to switch the order around several times and missed the QT5. I've pushed a change to the liblsl LSLCMake module so deploying works with Qt6, so once that's released Qt6 can be tried first by default.

File: .github/workflows/cppcmake.yml [hunk 2]
```diff
@@ -57,6 +53,16 @@ jobs:
            mkdir LSL
            tar -xvf liblsl.tar.bz2 -C LSL
 
+    - name: Install Qt
+      if: matrix.config.os == 'windows-latest' || matrix.config.os == 'macOS-latest'
```
[the remark this answers is not in the document]
Comment (tstenner) [reply .github/workflows/cppcmake.yml]: I have applied your sduggestion, but the negation doesn't make sense, as the `.deb` packages are only applicable to Debian / Ubuntu targets.

**Comment by tstenner** [comment]
It doesn't look pretty, the emulator doesn't support multicast and the folder selection might need some work so Android allows writing files to the storage. On the other hand it builds and launches:

Screenshot_1614708007-fs8

**Pull request merged**